### ISSUE TITLE
Archival node

### ISSUE THREAD

**Issue by ZenVoich** [kaspanet/rusty-kaspa#578]
I started integrating kaspa with rusty-kaspa because the old kaspad was deprecated. Now I need historical data.

Any plans on this?

**Comment by bunnyfu** [comment]
Safe to assume that GO nodes, if not unplugged, will remain on the unforked 1BPS network, so they will become irrelevant at that point.

**Comment by bunnyfu** [comment]
The real question is if anyone is working on archival nodes for rusty-kaspa! Per the Kaspa WIKI, you cannot run kaspad with the --archival parameter unless running the GO build.

**Comment by bunnyfu** [comment]
Yes, can't hardfork to 10BPS until archival nodes have been migrated to Rusty Kaspa. Otherwise there will be a pause in explorer data logging – a gap that must be avoided at this stage.

**Comment by bunnyfu** [comment]
@callid0n Thank you very much for this clarification. I may indeed have read the info wrong from the Wiki. I have not tried running a Rusty Kaspa archival node myself, because I read the Wiki as saying only GO nodes can have kaspad run with the --archival parameter, which brought me to the conclusion that this functionality is not present in Rusty Kaspa. So, it actually meant that kaspad on Rusty simply doesn't need this parameter? Then how do we put a Rusty node into archival mode?

**Comment by bobpinella** [comment]
I've been running a Rusty Kaspa node since October 2024 with the --archival flag.  Everything was working great until I upgraded to version 1.0.0 following the May 2025 network upgrade.  Now it seems to only be retaining archival data from the moment I started the new version.  All past archival data since October 2024 now appears to be gone because I cannot query any of those older blocks.  Anyone know how to properly run an archival node using version 1.0.0 while retaining older data?
